### ISSUE TITLE
Adding second attachment to an document on Cloudant fails

### ISSUE THREAD

**Issue by mstefaniuk** [pouchdb/pouchdb#5946]
### Issue
I'm using PouchDB as access API for Cloudant. I'm saving document and then attachment is added. Strangely first attachment is added with success but second save (without explicit document update) causes Cloudant error.

```erl
{
    error:"case_clause",
    reason:"{ok,[{ok,{doc,<<"AktrDbge">>,
              {2,[<<79,206,3,61,91,226,77,164,116,175,134,7,126,70,118,30>>]},
              {[{<<"type">>,<<"fiddle">>}]},
              [{att,<<"0.gv">>,<<"text/plain">>,48,31,
                    <<139,30,4,183,149,20,74,56,214,123,220,64,188,97,213,81>>,
                    2,
                    {<6067.14918.499>,[{3240151,48}]},
                    gzip}],
              false,[]}},
     {ok,{doc,<<"AktrDbge">>,
              {2,
               [<<79,206,3,61,91,226,77,164,116,175,134,7,126,70,118,30>>,
                <<169,158,125,189,14,54,29,195,63,60,199,6,183,248,52,112>>]},
              {[{<<"type">>,<<"fiddle">>}]},
              [{att,<<"0.gv">>,<<"text/plain">>,48,31,
                    <<139,30,4,183,149,20,74,56,214,123,220,64,188,97,213,81>>,
                    2,
                    {<6069.28175.3186>,[{4092119,48}]},
                    gzip}],
              false,[]}}]}",
    ref:4215018366
}
```
I suppose that it is PouchDB problem because it is working using `curl`.

### Info
- Environment: browser
- Platform: Chrome / FireFox
- Adapter: Remote
- Server: Cloudant

### Reproduce
Document in both attempts is saved using the same code.
```js
var attachment = new Blob([req.source], {type: 'text/plain'});
var count = currentAttachment(currentDocument._attachments);
var name = "" + (count + 1);
db.putAttachment(currentDocument._id, name + '.gv', currentDocument._rev, attachment, 'text/plain')
   .then(function (result) {
    ....
});
```
Debug log is as follows:
```
pouchdb:http "PUT https://graphvizit.cloudant.com/public/AktrDbge/0.gv?rev=1-a99e7dbd0e361dc33f3cc706b7f83470" +228mspouchdb.js:11255:8
pouchdb:http "GET https://graphvizit.cloudant.com/public/AktrDbge?" +133mspouchdb.js:11255:8
pouchdb:http "GET https://graphvizit.cloudant.com/public/AktrDbge/0.gv" +170mspouchdb.js:11255:8
pouchdb:http "PUT https://graphvizit.cloudant.com/public/AktrDbge/1.gv?rev=2-4fce033d5be24da474af86077e46761e" +3s
```
Working example you can check on http://graphviz.it. Save is working - update is not.

**Comment by nolanlawson** [comment]
Based on the Erlang stack trace you showed, this looks like an error in Cloudant rather than PouchDB. Is there any chance you could provide a _reduced_ test case to demonstrate the error? I see http://graphviz.it but I'm  not sure which buttons to click or what to look for to reproduce the error. A simple `index.html` file containing just the code necessary to reproduce the error would be extremely helpful. :smiley:

Also if you can confirm that this fails in Cloudant but works in CouchDB that would be another great piece of information. If that's the case then I'd advise checking CouchDB v1.6.1 vs CouchDB 2.0 to see that as well; Cloudant is largely based on CouchDB 2.0 although there is some custom code so knowing those three pieces of information would help Cloudant folks look into this. (And of course PouchDB can write a workaround if a workaround exists.)

**Comment by willholley** [comment]
@mstefaniuk this is a side effect of a CouchDB 2.0 [bug](https://issues.apache.org/jira/browse/COUCHDB-3097), I think. This should be fixed by https://github.com/apache/couchdb-fabric/pull/65 which we hope to get onto our servers in the next week or so.

**Comment by willholley** [comment]
@mstefaniuk looking at this further, it may be something different. Can you raise a ticket with Cloudant support please?

**Comment by mstefaniuk** [comment]
@willholley I've raised it twice with no result. You can check already closed #77064 and #77914.

**Comment by mstefaniuk** [comment]
Finally issue has been fixed on Cloudant side by moving from Amsterdam to London data center. It is not connected with PouchDB at all.
